spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java

```diff
@@ -283,14 +292,23 @@ public TaskExecutionResource.Page executionList() {
 	}
 
 	@Override
-	public PagedModel<TaskExecutionThinResource> thinExecutionList() {
+	public TaskExecutionThinResource.Page thinExecutionList() {
 		if(thinExecutionsLink != null) {
 			return restTemplate.getForObject(thinExecutionsLink.getHref(), TaskExecutionThinResource.Page.class);
 		} else {
 			return restTemplate.getForObject(executionsLink.getHref(), TaskExecutionThinResource.Page.class);
 		}
 	}
 
+	@Override
+	public TaskExecutionThinResource.Page thinExecutionListByTaskName(String taskName) {
+		if(thinExecutionsByNameLink != null) {
+			return restTemplate.getForObject(thinExecutionsByNameLink.expand(taskName).getHref(), TaskExecutionThinResource.Page.class);
+		} else {
```

Comment:
No need to have else, the last statement can just return.

Reply:
It's an old habit to have an else.